src/main/java/com/novauc/SpringBasicsFRController.java

```diff
@@ -0,0 +1,57 @@
+package com.novauc;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import javax.servlet.http.HttpSession;
+import java.util.ArrayList;
+
+@Controller
+public class SpringBasicsFRController {
+
+    public static ArrayList<Message> messagesAL = new ArrayList<>();
+
+    @RequestMapping(path = "/", method = RequestMethod.GET)
+    public String home(Model model, HttpSession session) {
+        model.addAttribute("name", session.getAttribute("userName"));
+        model.addAttribute("modeledMessage", messagesAL);
+        return "home";
+    }
+
+    @RequestMapping(path = "/login", method = RequestMethod.POST)
+    public String login(HttpSession session, String userName) {
+        session.setAttribute("userName", userName);
+        return "redirect:/";
+    }
+
+    @RequestMapping(path = "/add-message", method = RequestMethod.POST)
+    public String message(HttpSession session, String enterMessage) {
+        session.setAttribute("enterMessage", enterMessage);
+        messagesAL.add(new Message(enterMessage));
```

Comment:
Good job as always Peter.  Just a note you don't need the session related stuff in add-message and delete-message.  You are just adding those attributes to the session but you never pull them out for anything, so they are not needed.